### ISSUE TITLE
Syntax Error with for await

### ISSUE THREAD

**Issue by splitice** [javascript-obfuscator/javascript-obfuscator#419]
Example code:
```
const fs = require('fs'),
	  readline = require('readline')

function W(){
	
	async function lll(addr){
		const rli = readline.createInterface({
			input: fs.createReadStream("/proc/net/arp"),
			crlfDelay: Infinity
		});
		
		for await (let line of rli) {
			line = line.split(/\s+/)
			if(!line.length) continue
			if(line[0] == addr) return line[3]
		}
	}
	let macAddr = lll(device.address)
}
```
Obfustication code:
```
JavaScriptObfuscator.obfuscate(code,
            {
                compact: false,
                controlFlowFlattening: true,
                controlFlowFlatteningThreshold: 0.3,
                selfDefending: true,
                deadCodeInjection: true,
                deadCodeInjectionThreshold: 0.1,
                target: 'node'
            }
        )
```

Error:
```
missing ) after argument list
```

Result:
Results in the `forawait` instead of `for await` in the obfusticated code

**Comment by splitice** [comment]
Yield is also broken. The following code produces incorrect obfusticated code:

```
for (const value of this.all(p)) {
					yield { type: "entry", value };
					last = value;
				}
				yield { type: "complete", last };
				return;
```

Results in:
```
for (const _0x570726 of this['all'](_0x5e852b))
                        yield {
                            'type': 'entry',
                            'value': _0x570726
                        }, _0x4fdcc8 = _0x570726;
                    return void (yield {
                        'type': 'complete',
                        'last': _0x4fdcc8
                    });
```

That's wrong in multiple ways (comma can't follow yield and whats the deal with the void()?)

**Comment by sanex3339** [comment]
Please, can you provide a standalone example that will correctly evaluate inside the browser environment before obfuscation and will not - after obfuscation.

**Comment by sanex3339** [comment]
Got standalone example with await that produced error

```
async function issue419() {
    async function getStringPart(foundPart) {
        const promises = [
            Promise.resolve('foo foo'),
            Promise.resolve('bar bar')
        ];

        for await (let line of promises) {
            line = line.split(/\s+/)
            if (!line.length) continue
            if (line[0] == foundPart) return line[0]
        }
    }
    let stringPart = await getStringPart('bar')
    console.log(stringPart);
}

await issue419();
```

**Comment by sanex3339** [comment]
Downlevelled the problem with async to

```
            async function testFunc(params) {
                for await (let param of params) {
                  
                }
            }
            testFunc(['foo']);
```

Options:
```
        {
            ...NO_ADDITIONAL_NODES_PRESET,
            compact: true
        }
```

Looks like ` compact: true` removed space before `for` and `await`:

```
async function testFunc(_0x28503c){forawait(let _0x2304c3 of _0x28503c){}}testFunc(['foo']);
```

**Comment by splitice** [comment]
That's because `compact: true` causes a rewrite operation. The version in this repo doesnt correctly handle `for await` hence rewrites incorrectly. It's resolved by using the non patched upstream version of escodegen. Our fork now does just that.

**Comment by sanex3339** [comment]
I tried original `escodegen 1.12.0` but still error with `forawait`

**Comment by splitice** [comment]
https://github.com/HalleyAssist/javascript-obfuscator does not exhibit either problem. The only difference is the update to latest escodegen.

**Comment by sanex3339** [comment]
With `forawait`? I think it has, because the master branch has bug here
https://github.com/estools/escodegen/blob/master/escodegen.js#L956

`space` instead of `noEmptySpace()` between `for` and `await`

**Comment by splitice** [comment]
I believe so. Works closed until the 6th so I can't be 100%.

**Comment by sanex3339** [comment]
Created the PR:
https://github.com/estools/escodegen/pull/402

Please ping it

**Comment by sanex3339** [comment]
Cannot reproduce the second example with yield here:
https://obfuscator.io/

**Comment by sanex3339** [comment]
Fixed problem with for-await-of.
Cannot reproduce with yield.

Please, create another issue with code that will reproduce error with `yield` on https://obfuscator.io/

**Comment by sanex3339** [comment]
Released in `0.20.3`
